docs/examples/extensions/dashboard-section/js/index.js

```diff
@@ -0,0 +1,133 @@
+/** @format */
+/**
+ * External dependencies
+ */
+
+import { addFilter } from '@wordpress/hooks';
+import { Component, Fragment } from '@wordpress/element';
+import { TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * WooCommerce dependencies
+ */
+import { EllipsisMenu, MenuTitle, MenuItem, SectionHeader } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import UpcomingEvents from './upcoming-events';
+import GlobalPrices from './global-prices';
+
+const config = [
+	{
+		title: 'Granny Smith',
+		events: [ { date: '2020-01-02', title: 'The Granny Apple Fair' } ],
+	},
+	{
+		title: 'Golden Delicious',
+		events: [
+			{ date: '2020-04-15', title: 'Madrid Manzana Dorada' },
+			{ date: '2020-05-07', title: 'Golden CO Golden Delicious Day' },
+		],
+	},
+	{
+		title: 'Gala',
+		events: [ { date: '2020-08-31', title: 'The Met Gala Pomme' } ],
+	},
+	{
+		title: 'Braeburn',
+		events: [ { date: '2020-08-18', title: 'Mt. Aoraki Crisper' } ],
+	},
+];
+
+const dashboardItems = [
+	{ title: 'Upcoming Events', component: UpcomingEvents, key: 'upcoming-events' },
+	{ title: 'Global Apple Prices', component: GlobalPrices, key: 'global-prices' },
+];
+
+class Section extends Component {
+	renderMenu() {
+		const {
+			hiddenBlocks,
+			onToggleHiddenBlock,
+			onTitleBlur,
+			onTitleChange,
+			titleInput,
+			onMove,
+			onRemove,
+			isFirst,
+			isLast,
+			controls: Controls,
+		} = this.props;
+
+		return (
+			<EllipsisMenu
+				label={ __( 'Choose Apples', 'woocommerce-admin' ) }
+				renderContent={ ( { onToggle } ) => (
+					<Fragment>
+						<MenuTitle>{ __( 'My Apples', 'woocommerce-admin' ) }</MenuTitle>
+						{ dashboardItems.map( item => (
+							<MenuItem
+								checked={ ! hiddenBlocks.includes( item.key ) }
+								isCheckbox
+								isClickable
+								key={ item.key }
+								onInvoke={ () => onToggleHiddenBlock( item.key )() }
+							>
+								{ item.title }
+							</MenuItem>
+						) ) }
+						<div className="woocommerce-ellipsis-menu__item">
+							<TextControl
+								label={ __( 'Section Title', 'woocommerce-admin' ) }
+								onBlur={ onTitleBlur }
+								onChange={ onTitleChange }
+								required
+								value={ titleInput }
+							/>
+						</div>
+						<Controls
```

Comment:
Looking at this again... I wonder if it would make sense to move the title text input inside `<Controls>` so extension developers don't need to implement it. What do you think @psealock?

Reply:
I like that idea!

Reply:
Done in https://github.com/woocommerce/woocommerce-admin/pull/2280/commits/d8dd86f501fa7c652c0758faed84318c1adecfa2